components/admin/requests/processing/TasksCard.tsx

```diff
@@ -201,98 +211,131 @@ export default function ProcessingTasksCard({ applicationId }: ProcessingTasksCa
             >
               <Text textStyle="xsmall-medium">Mark as complete</Text>
             </Button>
-          )}
+          ) : null}
         </ProcessingTaskStep>
-        {/* Task 4: Review Request Information -> Disable Editing of First 3 steps */}
+        {/* Task 4: Review Information: Review Information (MODAL) */}
         <ProcessingTaskStep
           id={4}
-          label={`Review request information`}
+          label={'Review request information'}
           description="Editing will be disabled upon completion of this step"
-          isCompleted={hasReviewedRequestInformation}
+          isCompleted={reviewRequestTask}
         >
           <ReviewInformationModal
             applicationId={applicationId}
             requestType={'Request'}
-            onConfirmed={() => setHasReviewedRequest(true)}
+            onConfirmed={() => setReviewRequestTask(true)}
           >
-            {invoiceNumber === null ? (
+            {!reviewRequestTask ? (
               <Button
                 marginLeft="auto"
                 height="35px"
                 bg="background.gray"
                 _hover={{ bg: 'background.grayHover' }}
                 color="black"
+                disabled={!appNumber || !appHolepunched || !walletCardCreated}
               >
-                <Text textStyle="xsmall-medium">Review Information</Text>
+                <Text textStyle="xsmall-medium">Review information</Text>
               </Button>
             ) : (
-              <Button variant="ghost" textDecoration="underline black">
+              <Button
+                variant="ghost"
+                textDecoration="underline black"
+                onClick={() => {
+                  setDocumentUploadTask(false);
+                  setGenerateInvoiceTask(false);
+                  handleMailOut(false);
+                  setReviewRequestTask(false);
+                }}
+              >
                 <Text textStyle="caption" color="black">
-                  Edit number
+                  Undo Review
                 </Text>
               </Button>
             )}
           </ReviewInformationModal>
         </ProcessingTaskStep>
-        {/* Task 5: Assign invoice number: Assign number (MODAL) */}
+        {/* Task 5: Generate Invoice */}
         <ProcessingTaskStep
           id={5}
-          label={`Assign invoice number${invoiceNumber === null ? '' : `: ${invoiceNumber}`}`}
-          description="Include permit number, expiry date, full name, and birth month"
-          isCompleted={invoiceNumber !== null}
+          label="Generate Invoice"
+          description="Invoice number will be automatically assigned"
+          isCompleted={generateInvoiceTask && reviewRequestTask}
```

Comment:
```suggestion
          isCompleted={generateInvoiceTask}
```
Logically, whether this step is complete should only depend on the boolean state of `generateInvoiceTask`. You also set `generateInvoiceTask` to `false` when undoing the review step